src/components/services/FioService.ts

```diff
@@ -0,0 +1,65 @@
+import { EdgeCurrencyWallet } from 'edge-core-js'
+import * as React from 'react'
+
+import { checkExpiredFioDomains } from '../../actions/FioActions'
+import { useDispatch, useSelector } from '../../types/reactRedux'
+import { NavigationBase } from '../../types/routerTypes'
+import { FioDomain } from '../../types/types'
+import { needToCheckExpired, refreshFioNames } from '../../util/FioAddressUtils'
+import { makePeriodicTask } from '../../util/PeriodicTask'
+
+const EXPIRE_CHECK_TIMEOUT = 30000
+
+interface Props {
+  navigation: NavigationBase
+}
+
+export const FioService = (props: Props) => {
+  const { navigation } = props
+  const dispatch = useDispatch()
+
+  const expiredLastChecks = useSelector(state => state.ui.fio.expiredLastChecks)
+  const expireReminderShown = useSelector(state => state.ui.fio.expireReminderShown)
+  const expiredChecking = useSelector(state => state.ui.fio.expiredChecking)
+  const walletsCheckedForExpired = useSelector(state => state.ui.fio.walletsCheckedForExpired)
+  const fioWallets = useSelector(state => state.ui.wallets.fioWallets)
+
+  // Check for expired FIO domains
+  React.useEffect(() => {
+    const task = makePeriodicTask(async () => {
+      if (expireReminderShown) return
+
+      if (fioWallets.length === 0) {
+        return
+      }
+
+      if (expiredChecking) return
+
+      const walletsToCheck: EdgeCurrencyWallet[] = []
+      for (const fioWallet of fioWallets) {
+        if (!walletsCheckedForExpired[fioWallet.id]) {
+          walletsToCheck.push(fioWallet)
+        }
+      }
+
+      const namesToCheck: FioDomain[] = []
+      const { fioDomains, fioWalletsById } = await refreshFioNames(walletsToCheck)
+      for (const fioDomain of fioDomains) {
+        if (needToCheckExpired(expiredLastChecks, fioDomain.name)) {
+          namesToCheck.push(fioDomain)
+        }
+      }
+
+      if (namesToCheck.length !== 0) {
+        dispatch({ type: 'FIO/CHECKING_EXPIRED', data: true })
+        await dispatch(checkExpiredFioDomains(navigation, namesToCheck, fioWalletsById))
+      }
+    }, EXPIRE_CHECK_TIMEOUT)
+    task.start()
+
+    return () => task.stop()
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
We don't want to stop & start the task a bunch of times, but we also don't want the loop to be stale.

So, I would do this:

```ts
const refreshNamesToCheckExpired = useHandler(()= > {
  ... the body ...
})

React.useEffect(() => {
   makePeriodicTask(
     refreshNamesToCheckExpired,
     {
      onError: ....
     }
  })
  task.start()

  return () => task.stop()
})
```

That means the `refreshNamesToCheckExpired` function will have access to the fresh values, but the periodic task will keep its identity.